extensions/csrc/cuda/rms_layernorm_kernel.cu

```diff
@@ -12,6 +12,34 @@
 #include "../common/micros.h"
 #include "../common/cuda_type_utils.h"
 
+#define DISPATCH_RMSNORM_FLOAT_HALF_AND_BFLOAT(DATA_SIZE, TYPE, NAME, ...)  \
+  if (DATA_SIZE == 2) {                                                     \
```

Comment:
Why do we use if here

Reply:
Here we'll invoke different kernels based on its data type. Let's say for fp16/bf16/int4, GPU supports vectorized elementwise operations like __hadd2, __hmul2, thus we'll invoke `fused_add_rms_layernorm_kernel` kernel function to enable the vectorized computation, but for other data types like float, GPU doesn't support any vectorized operations, thus we fall back to `general_fused_add_rms_layernorm_kernel`